### DISCUSSION TITLE
get username in app.OnRecordAfterCreateRequest

### DISCUSSION THREAD

**Discussion by dankeyy** [pocketbase/pocketbase#3124] [Q&A] [answered]
So I have this hook, and I want to be able to do a certain operation on the backend based on who's the user making the request.

I see that the token is sent in the Authorization field header but I'm unsure how to get the username out of it and/or if this is the recommended way to do so
Thanks

**Comment by ganigeorgiev** [comment, marked as answer]
You can check the "Retrieving the current auth state" guide:

- Go: https://pocketbase.io/docs/go-routing/#retrieving-the-current-auth-state
- JS: https://pocketbase.io/docs/js-routing/#retrieving-the-current-auth-state

_(the hook `e.HttpContext` field is `c` in the above examples)_

Here is an example:

```go
// Go
app.OnRecordAfterCreateRequest("yourCollection").Add(func(e *core.RecordCreateEvent) error {
	// note: depending on you Collection API rule, this could be nil
	// OR
	// performed by admin (aka. e.HttpContext.Get(apis.ContextAdminKey).(*models.Admin))
	record, _ := e.HttpContext.Get(apis.ContextAuthRecordKey).(*models.Record)

	if record != nil {
		log.Println(record.Username())
	}

	return nil
})
```

```js
// JS
onRecordAfterCreateRequest((e) => {
    // note: depending on you Collection API rule, this could be null
    // OR
    // performed by admin (aka. e.httpContext.get("admin"))
    const authRecord = e.httpContext.get("authRecord")

    console.log(authRecord?.username())
}, "yourCollection")
```

**Reply by dankeyy** [reply]
perfect, thanks Gani!
